cpp/net/connection_pool.cc

```diff
@@ -16,45 +17,98 @@ DEFINE_int32(url_fetcher_max_conn_per_host_port, 4,
 namespace cert_trans {
 namespace internal {
 
+ConnectionPool::Connection::Connection(evhtp_connection_t* conn,
+                                       HostPortPair&& other_end)
+    : conn_(CHECK_NOTNULL(conn)), other_end_(other_end) {
+}
+
+
+const HostPortPair& ConnectionPool::Connection::other_end() const {
+  return other_end_;
+}
+
+
+void ConnectionPool::Connection::ReleaseConnection() {
+  VLOG(1) << "Releasing connection " << conn_.get();
+  conn_.release();
+}
+
+
 ConnectionPool::ConnectionPool(libevent::Base* base)
     : base_(CHECK_NOTNULL(base)), cleanup_scheduled_(false) {
 }
 
 
-evhttp_connection_unique_ptr ConnectionPool::Get(const URL& url) {
+namespace {
+
+
+evhtp_res ConnectionClosedHook(evhtp_connection_t* conn, void* arg) {
+  CHECK_NOTNULL(conn);
```

Comment:
By "rolling it in", I meant that you wouldn't need a helper method to do `conn_.reset()`, it could just *do it*.